truss/truss_config.py

```diff
@@ -372,3 +349,52 @@ def clone(self):
     def validate(self):
         for secret_name in self.secrets:
             validate_secret_name(secret_name)
+
+
+dataclass_to_req_keys_map = {
+    Train: {"variables"},
+    Resources: {"accelerator", "cpu", "memory", "use_gpu"},
+    TrussConfig: {
+        "environment_variables",
+        "external_package_dirs",
+        "model_metadata",
+        "model_name",
+        "python_version",
+        "requirements",
+        "resources",
+        "secrets",
+        "system_packages",
+    },
+    BaseImage: {"image", "python_executable_path"},
+}
+
+types_to_serialize = (Resources, Train, BaseImage)
```

Comment:
hows this different than the keys of the object right above? Might be good to write some comments on how you intend to use these

Reply:
we can just use the keys from the object above